Update Documentation: document APPLICATION_NAME_OPTS environment variable (#38001)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/reference/platforms/jvm/application_plugin.adoc

The start scripts launch a JVM with the specified settings defined as part of the original build and runtime environment (e.g. `JAVA_OPTS` env var).
The default script templates are based on the same scripts used to launch Gradle itself, that ship as part of a Gradle distribution.

The generated start scripts also support the `<APPLICATION_NAME>_OPTS` environment variable for specifying additional JVM arguments at runtime.

> 

Reviewer: ```suggestion
The application name is converted to uppercase with non-alphanumeric characters replaced by underscores.
```

> For example, if your application name is `my-app`, you can use:

Reviewer: ```suggestion
For example, an application named `my-app` or `myApp` both use `MY_APP_OPTS`:
```
